crates/lib/src/cli.rs

```diff
@@ -975,7 +985,11 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
             }
         }
     } else {
-        let fetched = crate::deploy::pull(repo, imgref, None, opts.quiet, prog.clone()).await?;
+        let fetched = if opts.unified {
+            crate::deploy::pull_unified(repo, imgref, None, opts.quiet, prog.clone(), sysroot).await?
+        } else {
+            crate::deploy::pull(repo, imgref, None, opts.quiet, prog.clone()).await?
```

Comment:
This relates to https://github.com/bootc-dev/bootc/pull/1599#discussion_r2333604990

Basically how about changing this code to take a `Storage` which would hold the persistent flag, and then `crate::deploy::pull` would itself query that flag and change its behavior.

That would also implicitly then fix the install path to behave the same.